src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -0,0 +1,711 @@
+
+  auto countCell = [&](localIndex c, std::map<int,std::vector<vtkIdType >> & regionsElem ) { // auto can be multiple types in the same function
+    if (attributeDataArray != nullptr)
+    {
+      int region = attributeDataArray->GetValue(c);
+      GEOSX_LOG_RANK_0_IF( region < 0,"Attribute value " << region << " not supported, please use value >=0 to describe regions");
+      regionsElem[region].push_back(c);
+    }
+    else
+    {
+      regionsElem[-1].push_back(c);
+    }
+  };
+
+
+  for( vtkIdType c = 0; c < mesh.GetNumberOfCells(); c++)
+  {
+    if( mesh.GetCellType(c) == VTK_HEXAHEDRON )
+    {
+      countCell(c, regionsHex);
+    }
+    else if( mesh.GetCellType(c) == VTK_TETRA )
+    {
+      countCell(c, regionsTetra);
+    }
+    else if( mesh.GetCellType(c) == VTK_WEDGE )
+    {
+      countCell(c, regionsWedges);
+    }
+    else if( mesh.GetCellType(c) == VTK_PYRAMID )
+    {
+      countCell(c, regionsPyramids);
+    }
+    else if( mesh.GetCellType(c) == VTK_TRIANGLE || mesh.GetCellType(c) == VTK_QUAD )
+    {
+      if (attributeDataArray != nullptr)
+      {
+        surfaces[attributeDataArray->GetValue(c)].push_back(c);
+      }
+    }
+  }
+
+  // Fill the region set
+  auto fillRegionSet = [&](std::map<int,std::vector<vtkIdType >> & regionsElem ) {
+  for( auto const & region : regionsElem )
+  {
+    regions.insert(region.first);
+  }
+  };
+  fillRegionSet(regionsHex);
+  fillRegionSet(regionsTetra);
+  fillRegionSet(regionsPyramids);
+  fillRegionSet(regionsWedges);
+
+
+  // Communicate all the surfaces attributes to the ranks
+  std::vector< int > surfaceVector;
+  surfaceVector.reserve(surfaces.size());
+  for( auto & surface : surfaces)
+  {
+    surfaceVector.push_back( surface.first );
+  }
+  array1d< int > surfaceSizes( MpiWrapper::commSize());
+  MpiWrapper::allGather( LvArray::integerConversion<int>(surfaceVector.size()), surfaceSizes, MPI_COMM_GEOSX );
+  int const totalNbSurfaceId = std::accumulate( surfaceSizes.begin(), surfaceSizes.end(), 0 );
+  GEOSX_LOG_RANK("total nb surface id " << totalNbSurfaceId);
+  allSurfaces.resize(totalNbSurfaceId);
+  std::vector< int > displacements( MpiWrapper::commSize(),0);
+  std::partial_sum( surfaceSizes.begin(), surfaceSizes.end() - 1, displacements.begin() + 1 );
+  MpiWrapper::allgatherv(surfaceVector.data(),  surfaceVector.size(), allSurfaces.data(),surfaceSizes.data(), displacements.data(), MPI_COMM_GEOSX);
+
+  std::sort(allSurfaces.begin(), allSurfaces.end());
+  auto last = std::unique( allSurfaces.begin(), allSurfaces.end());
+  allSurfaces.erase(last, allSurfaces.end());
+}
+
+/**
+ * @brief Find the properties to be imported
+ * @details all the float and double vtkArrays will be imported
+ * @return a vector containing all the vtkDataArray that can be imported
+ */
+std::vector< vtkDataArray * > findArrayToBeImported(vtkUnstructuredGrid & mesh)
```

Comment:
I think @TotoGaz already mentioned this, but these arrays should be const since we're not modifying them
```suggestion
std::vector< vtkDataArray const * > findArrayToBeImported(vtkUnstructuredGrid & mesh)
```